File: NetFramework/DataFilter/VB/ChartAsDataFilter/ChartAsDataFilter/Data/DataService.vb
Imports System.IO
Imports System.Web.Script.Serialization

Namespace ChartAsDataFilter.Data
    Public Class DataService
        Private Shared dt As List(Of Quote)

        Shared Sub New()
            Dim buildDir = Path.GetDirectoryName(System.Reflection.Assembly.GetExecutingAssembly().Location)
            Dim quotesPath = Path.Combine(buildDir, "Resources", "HistoricalQuotes.json")
            Dim text = File.ReadAllText(quotesPath)
            Dim jsonSerializer = New JavaScriptSerializer()
            dt = jsonSerializer.Deserialize(Of List(Of Quote))(text)
        End Sub

        Public Shared Function GetFinancialData(ByVal Optional count As Integer = 365) As List(Of Quote)
            count = Math.Min(count, dt.Count)
            Return dt.Take(count).OrderBy(Function(x) x.Day).ToList()
        End Function

        Public Shared Function GetFinancialData(ByVal startDate As DateTime, ByVal Optional count As Integer = 365) As List(Of Quote)
            Dim quotes = GetFinancialData(count)

            For Each quote In quotes
                quote.Day = startDate.AddDays(quotes.IndexOf(quote))
            Next

            Return quotes
        End Function
    End Class
End Namespace

File: NetFramework/DataFilter/CS/ChartAsDataFilter/ChartAsDataFilter/Data/DataService.cs
using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.IO;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Web.Script.Serialization;

namespace ChartAsDataFilter.Data
{
    public class DataService
    {
        private static List<Quote> dt;
        static DataService()
        {
            var buildDir = Path.GetDirectoryName(System.Reflection.Assembly.GetExecutingAssembly().Location);
            var quotesPath = Path.Combine(buildDir, "Resources", "HistoricalQuotes.json");
            var text = File.ReadAllText(quotesPath);
            var jsonSerializer = new JavaScriptSerializer();
            dt = jsonSerializer.Deserialize<List<Quote>>(text);
        }        
        public static List<Quote> GetFinancialData(int count = 365)
        {
            count = Math.Min(count, dt.Count);
            return dt.Take(count).OrderBy(x => x.Date).ToList();
        }

        public static List<Quote> GetFinancialData(DateTime startDate, int count = 365)
        {
            var quotes = GetFinancialData(count);
            foreach (var quote in quotes)
            {
                quote.Date = startDate.AddDays(quotes.IndexOf(quote));
            }
            return quotes;
        }
    }
}
